contracts/AllowedStrategy.sol

```diff
@@ -0,0 +1,270 @@
+
+    /**
+     * @notice Used when attempting to deposit when there are not active groups
+     * to vote for.
+     */
+    error NoActiveGroups();
+
+    /**
+     * @notice Initialize the contract with registry and owner.
+     * @param _registry The address of the Celo registry.
+     * @param _owner The address of the contract owner.
+     * @param _manager The address of the Manager contract.
+     */
+    function initialize(
+        address _registry,
+        address _owner,
+        address _manager
+    ) external initializer {
+        _transferOwnership(_owner);
+        __UsingRegistry_init(_registry);
+        __Managed_init(_manager);
+    }
+
+    /**
+     * @notice Set this contract's dependencies in the StakedCelo system.
+     * @param _account The address of the Account contract.
+     * @param _groupHealth The address of the Group health contract.
+     */
+    function setDependencies(address _account, address _groupHealth) external onlyOwner {
+        require(_account != address(0), "account null address");
+        require(_groupHealth != address(0), "validate group null address");
+
+        account = IAccount(_account);
+        groupHealthContract = IGroupHealth(_groupHealth);
+    }
+
+    /**
+     * @notice Marks a group as allowed strategy for voting.
+     * @param group The address of the group to add to the set of allowed
+     * strategies.
+     */
+    function allowStrategy(address group) external onlyManager {
+        if (!groupHealthContract.isValidGroup(group)) {
+            revert StrategyNotEligible(group);
+        }
+
+        if (allowedStrategies.contains(group)) {
+            revert StrategyAlreadyAdded(group);
+        }
+
+        if (!allowedStrategies.add(group)) {
+            revert FailedToAddAllowedStrategy(group);
+        }
+
+        emit StrategyAllowed(group);
+    }
+
+    /**
+     * @notice Marks a group as not allowed strategy for voting
+     * and redistributes votes to default strategy.
+     * @param group The address of the group to remove from the set of allowed
+     * strategies.
+     * @return total stCelo in blocked strategy
+     */
+    function blockStrategy(address group) external onlyManager returns (uint256) {
+        if (IManager(manager).getGroupsLength() == 0) {
+            revert NoActiveGroups();
+        }
+
+        if (!allowedStrategies.contains(group)) {
+            revert StrategyAlreadyBlocked(group);
+        }
+
+        if (!allowedStrategies.remove(group)) {
+            revert FailedToBlockStrategy(group);
+        }
+
+        emit StrategyBlocked(group);
+
+        return getTotalStCeloVotesForStrategy(group);
+    }
+
+    /**
+     * @notice Used to withdraw CELO from the system from allowed strategy
+     * that account voted for previously. It is expected that strategy will be balanced.
+     * For balancing use `rebalance` function
+     * @param strategy The validator group that we want to withdraw from.
```

Comment:
Is this a general strategy or a specific group? The name implies the former, the comment the latter.